core/dbt/parser/manifest.py

```diff
@@ -988,10 +988,15 @@ def build_manifest_state_check(self):
             env_var_str += f"{key}:{config.profile_env_vars[key]}|"
         profile_env_vars_hash = FileHash.from_contents(env_var_str)
 
-        # Create a FileHash of the profile file
-        profile_path = os.path.join(get_flags().PROFILES_DIR, "profiles.yml")
-        with open(profile_path) as fp:
-            profile_hash = FileHash.from_contents(fp.read())
+        # Create a hash of the connection keys, which user has access to in
+        # jinja context. Thus keys here may affect the parsing result.
+
+        # Renaming this variable mean that we will have to do a whole lot more
+        # change to make sure the previous manifest can be loaded correctly.
+        # This is an example of naming should be chosen based on the functionality
+        # rather than the implementation details.
+        connection_keys = config.credentials._connection_keys()
```

Comment:
It looks like `credentials._connection_keys()` is just the set of names of credential keys, e.g: 
```python
('account', 'user', 'database', 'warehouse', 'role', 'schema', 'authenticator', 'oauth_client_id', 'query_tag', 'client_session_keep_alive', 'host', 'port', 'proxy_host', 'proxy_port', 'protocol', 'connect_retries', 'connect_timeout', 'retry_on_database_errors', 'retry_all', 'insecure_mode', 'reuse_connections')
```

If we hash just the key names without the values, then partial parsing won't retrigger even if one of the values change, as it should. We should use `list(config.credentials.credential_info())` instead. `with_aliases=False` looks appropriate here as well.

Reply:
Good catch! This also mean I probably should add an integration test to capture the behavior

Reply:
You beat me to my suggestion 😂